From 5c6f22c0b279927cd8f185f0eba9f311131e092e Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 28 May 2023 00:29:09 +1100
Subject: [PATCH 2/3] don't pin find_package(fmt) nor swallow logs

---
 third_party/fmt/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/fmt/CMakeLists.txt b/third_party/fmt/CMakeLists.txt
index 373faaf..5b95f1f 100644
--- a/third_party/fmt/CMakeLists.txt
+++ b/third_party/fmt/CMakeLists.txt
@@ -1,6 +1,6 @@
 
 message(STATUS "Looking for fmt dependency")
-find_package(fmt 6.1 QUIET CONFIG)
+find_package(fmt CONFIG)
 if (fmt_FOUND)
     message(STATUS "Looking for fmt dependency -- found")
     add_custom_target(fmt_dependency)
